src/events.py

```diff
@@ -41,10 +43,18 @@ def __init__(self, **kw):
         for key in kw:
             setattr(self, key, kw[key])
 
-    def __cmp__(self, other):
-        if self.tick < other.tick: return -1
-        elif self.tick > other.tick: return 1
-        return cmp(self.data, other.data)
+    def __lt__(self, other):
+        if self.tick < other.tick:
+            return True
+        return self.data < other.data
```

Comment:
You should be defining all six rich comparison operators, or using the `functools.total_ordering` class decorator to get them.